tck/src/main/java/org/eclipse/microprofile/concurrency/tck/cdi/CDIBean.java

```diff
@@ -190,4 +252,251 @@ public void checkMaxQueued3(ManagedExecutor executor) throws ExecutionException,
+            Label.set("testInjectThreadContextNoConfig-label");
+            Buffer.set(new StringBuffer("testInjectThreadContextNoConfig-buffer"));
+
+            Function<Integer, String> testAllContextPropagated = defaultContext.contextualFunction(i -> {
+                Assert.assertEquals(Buffer.get().toString(), "testInjectThreadContextNoConfig-buffer",
+                        "Thread context type (Buffer) was not propagated.");
+                Assert.assertEquals(Label.get(), "testInjectThreadContextNoConfig-label",
+                        "Thread context type (Lable) was not propagated.");
+                Assert.assertEquals(Thread.currentThread().getPriority(), newPriority,
+                        "Thread context type (ThreadPriority) was not propagated.");
+                return "done";
+            });
+
+            Thread.currentThread().setPriority(4);
+            Label.set("testInjectThreadContextNoConfig-new-label");
+            Buffer.set(new StringBuffer("testInjectThreadContextNoConfig-new-buffer"));
+
+            testAllContextPropagated.apply(100);
+        }
+        finally {
+            // restore previous values
+            Buffer.set(null);
+            Label.set(null);
+            Thread.currentThread().setPriority(originalPriority);
+        }
+    }
+
+    /**
+     * When the container creates a ThreadContext for an injection point with an empty ThreadContextConfig annotation,
+     * its configuration is equivalent to ThreadContext.builder().build(), which is that all available thread context
+     * types are propagated, except for Transaction context, which is cleared.
+     */
+    public void testInjectThreadContextEmptyConfig() throws Exception {
+        int originalPriority = Thread.currentThread().getPriority();
+        int newPriority = originalPriority == 2 ? 1 : 2;
+        try {
+            Thread.currentThread().setPriority(newPriority);
+            Label.set("testInjectThreadContextEmptyConfig-label");
+            Buffer.set(new StringBuffer("testInjectThreadContextEmptyConfig-buffer"));
+
+            Callable<Boolean> testAllContextPropagated = defaultContext.contextualCallable(() -> {
+                Assert.assertEquals(Buffer.get().toString(), "testInjectThreadContextEmptyConfig-buffer",
+                        "Thread context type (Buffer) was not propagated.");
+                Assert.assertEquals(Label.get(), "testInjectThreadContextEmptyConfig-label",
+                        "Thread context type (Lable) was not propagated.");
+                Assert.assertEquals(Thread.currentThread().getPriority(), newPriority,
+                        "Thread context type (ThreadPriority) was not propagated.");
+                return true;
+            });
+
+            Thread.currentThread().setPriority(4);
+            Label.set("testInjectThreadContextNoConfig-new-label");
+            Buffer.set(new StringBuffer("testInjectThreadContextNoConfig-new-buffer"));
```

Comment:
Copy/paste of "NoConfig" instead of using "EmptyConfig"?

Reply:
Good catch. This doesn't effect the correctness of the test (it tests that a previously captured context is used instead), however, it will be much clearer to use consistent names.  Fixing...